### DISCUSSION TITLE
error with findAlleles command

### DISCUSSION THREAD

**Discussion by NoELJ22** [milaboratory/mixcr#1048] [Q&A] [answered]
Hello,

We are doing a BCR and TCR analysis on human data using the NEBNext® Immune Sequencing and I have an error at the Allele inference step with mixcr 4.3.

The upstream analysis worked well with the `nebnext-human-tcr-bcr-full-length `preset. But I have an error with de `findAlleles command` :
![error_findAlleles](https://user-images.githubusercontent.com/115169629/226668622-3fe4d3b7-34a3-4b02-9ebe-9cfc68974200.png)

Do you know where this error could come from ?

With the  `nebnext-human-tcr-full-length `and `nebnext-human-bcr-full-length ` presets I didn't have this error with mixcr 4.2, but i had the same error with mixcr 4.3 .

My code:

```
mixcr analyze nebnext-human-tcr-bcr-full-length\
   --tag-pattern '^N{7}(R1:*) \ ^(UMI:N{12})N{4}(R2:*)' \
   --rna \
   --remove-step exportClones \
   NEB1-1995-100BT_S5_L001_R1_001.fastq.gz NEB1-1995-100BT_S5_L001_R2_001.fastq.gz NEB1-1995-100BT_S5 

mixcr exportReports NEB1-1995-100BT_S5.clns NEB1-1995-100BT_S5.report.txt

mixcr exportQc align NEB1-1995-100BT_S5.clns alignQcclns.pdf
mixcr exportQc align NEB1-1995-100BT_S5.vdjca alignQcvdjca.pdf
mixcr exportQc chainUsage NEB1-1995-100BT_S5.clns chainUsage.pdf


mixcr findAlleles --report result_findAlleles/NEB1-1995-100BT.findAlleles.report.txt \
--json-report result_findAlleles/NEB1-1995-100BT.findAlleles.report.json \
--export-alleles-mutations NEB1-1995-100BT_alleles.tsv \ 
--output-template NEB1-1995-100BT_S5.reassigned.clns NEB1-1995-100BT_S5.clns 

mixcr exportClones NEB1-1995-100BT_S5.reassigned.clns  NEB1-1995-100BT_S5.mutation.tsv -allNMutations FR1Begin FR3End -nMutations FR4 -chains -cGene --filter-stops -vAlignment --filter-out-of-frames


mixcr findShmTrees \
   --report shm_result/NEB1-1995-100BT_S5_trees.log \
   NEB1-1995-100BT_S5.reassigned.clns \
   shm_result/NEB1-1995-100BT_S5.shmt
   

 mixcr exportShmTrees \
   shm_result/NEB1-1995-100BT_S5.shmt \
   tree_result/NEB1-1995-100BT_S5_trees.tsv 


mixcr exportShmTreesWithNodes \
   shm_result/NEB1-1995-100BT_S5.shmt \
   tree_result/NEB1-1995-100BT_S5_trees_nodes.tsv 
```

I attach the report for my sample
[NEB1-1995-100BT_S5.report.txt](https://github.com/milaboratory/mixcr/files/11030848/NEB1-1995-100BT_S5.report.txt)

Thank you for your help

**Comment by mizraelson** [comment]
Hi, thanks for the report, that seems to be a known bug which is already fixed in the dev version, which you can download via the following link:

https://link.milaboratory.com/software/mixcr/mixcr-develop.zip

Please let me know if it works.

Meanwhile, is there a reason why you specify the pattern `--tag-pattern '^N{7}(R1:*) \ ^(UMI:N{12})N{4}(R2:*)'`, if you are using NEB kit the pattern is already included in the preset and it differs from the one you use: 

`^N{22}(R1:*) \ ^(UMI:N{17})N{10}(R2:*)`

Also, there is no need to specify `--rna` as its already included.

I have also looked at the report and it mostly looks fine, but it seems that the sequencing quality dropped low at the end of the second read (which is a common case with 300+300), that's why it was trimmed by on average 27 bp:

```
Trimming report:
  R1 reads trimmed left: 224 (0.02%)
  R1 reads trimmed right: 227302 (24.88%)
  Average R1 nucleotides trimmed left: 0.03396903029759025
  Average R1 nucleotides trimmed right: 1.5750215097784657
  R2 reads trimmed left: 468 (0.05%)
  R2 reads trimmed right: 728608 (79.76%)
  Average R2 nucleotides trimmed left: 0.03046507096584926
  Average R2 nucleotides trimmed right: 27.645503525852234
```

That might be a possible reason for the number of overlapped reads which should be higher:

`Overlapped: 731723 (80.1%)`

And most likely because of that there are losses during clonotype assembly, because the reads that do not overlap don't fully cover the assembling feature (which is a full `VDJRegion` in that case):

```
Reads dropped due to the lack of a clone sequence, percent of total: 164684 (18.03%)
```
Everything else seems to be fine. 

Also, `nebnext-human-tcr-bcr-full-length` under the hood is the same as `nebnext-human-bcr-full-length `. Did you use the same MiXCR version in both cases (old and new preset)?

**Reply by NoELJ22** [reply]
Hello,

Thank you for your answer.

Indeed, the `--tag-pattern '^N{7}(R1:*) \ ^(UMI:N{12})N{4}(R2:*)' `option was a mistake. So I restarted my code with the dev version, removing the `--tag-pattern `and `-rna` options.
But I still have an error : 
![error_findAlleles](https://user-images.githubusercontent.com/115169629/227174342-48764ec7-92d8-4e63-9801-45d3425fd413.png)


For the quality control, indeed, we can see that the R2 sequencing quality is not so good with FASTQC. We are still working on the optimization of the experiment.

For the preset `nebnext-human-bcr-full-length` and `nebnext-human-tcr-full-length` I had launched it with mixcr4.2, but indeed they don't work either with mixcr4.3 and the  dev version.

**Comment by mizraelson** [comment]
Hi, it seems like the update didn't get through at the moment you downloaded it. Can you please try using the same link again (the version has been updated).
Sorry about that and let me know if it works.

**Comment by NoELJ22** [comment, marked as answer]
Hello,
It works. 
Thank you for your help

**Reply by mizraelson** [reply]
Happy to help!
